Merge Develop and Staging to keep it up to date

aanchirinah opened the pull request.

aanchirinah committed:
Merge pull request #3 from andela-aanchirinah/staging

Staging

aanchirinah committed:
Merge pull request #6 from andela-aanchirinah/staging

Staging

Avuidrauxs committed:
style(scss)
git commit -m

aanchirinah committed:
Merge pull request #7 from andela-aanchirinah/checkpoint/feedback/fix-article-image-heights

Checkpoint 1 Review

Avuidrauxs committed:
style(NewsHome): display news source title above articles
- changed action method to fetch an object with both article array and source name
[fixed]

aanchirinah committed:
Merge pull request #8 from andela-aanchirinah/checkpoint/feedback/display-newssource-title-above-articles

Checkpoint 1 Review

Avuidrauxs committed:
style(Login.jsx): fix display of content in the middle
- editted the scss that handles the style of Login.jsx
- justified content in the center of the page
[fixed]

aanchirinah committed:
Merge pull request #9 from andela-aanchirinah/checkpoint/feedback/align-login-elements-to-center

Checkpoint 1 Review

Avuidrauxs committed:
fix(.gitignore): gitignore was not working properly
- it is now working correctly
[fixed]

Avuidrauxs committed:
fix(.gitignore): gitignore was not working properly
- it is now working correctly
[fixed]

Avuidrauxs committed:
feat(tests): implement tests for my components and functions
- added test for all components
[delivered]

Avuidrauxs committed:
corrected .gitignore

aanchirinah committed:
Merge pull request #10 from andela-aanchirinah/checkpoint/feedback/flux-tests

Checkpoint 1 Review

Avuidrauxs committed:
corrected .gitignore

Avuidrauxs committed:
Deployment to Heroku

Avuidrauxs committed:
Deployment to Heroku

Avuidrauxs committed:
Deployment to Heroku

Avuidrauxs committed:
Deployment to Heroku

Avuidrauxs committed:
Deployment to Heroku

Avuidrauxs committed:
feat(webpack.config.js): implement a better way to load sass files
- added a loader in webpack to handle scss modules
- also fixed style margins in Nav.js
[delivered]

Avuidrauxs committed:
feat(webpack.config.js): implement a better way to load sass files
- added a loader in webpack to handle scss modules
- also fixed style margins in Nav.js
[delivered]

Avuidrauxs committed:
refactor(fluxActions): refactor action functions to arrow functions
- refactored the functions for each action to es 6 arrow functions
[delivered]

aanchirinah committed:
Merge pull request #12 from andela-aanchirinah/checkpoint/feeback/fix-sass-loader

Checkpoint/feeback/fix sass loader

Avuidrauxs committed:
feat(fluxActions.jsx): implement dynamic processes in functions
- removed hardcoded sort by parameters
- fixed radio button in component view
[delivered]

Avuidrauxs committed:
feat(social sharing): implement social sharing
[delivered]

aanchirinah committed:
Merge pull request #14 from andela-aanchirinah/checkpoint/add-social-share-buttons

Checkpoint 1.4 Review

amos-o commented on app/components/Nav.jsx:
```diff
@@ -16,34 +16,36 @@ class Nav extends React.Component {
   checkToken() {
     const tempToken = localStorage.getItem('jwtToken');
-    if (tempToken) {
+    console.log(tempToken, 'token');
```
Remove log statements from code

amos-o commented on app/styles/components/_inline-radio.scss:
```diff
@@ -0,0 +1,7 @@
+
+.inline-radio {
+	display: flex;
```
Fix indentation

amos-o commented on app/styles/components/_center-login.scss:
```diff
@@ -4,4 +4,15 @@
+
+.menu-left {
+  margin-left: 28%;
+    width: 100%;
```
Fix indent

amos-o commented on app/action/fluxActions.jsx:
```diff
@@ -18,45 +18,62 @@ export function fetchAllNewsSources() {
-}
+  // } else {
+  //   axios
+  //   .get(`${process
```
Remove commented code from the file if it is not being used

olawalejarvis commented on app/action/fluxActions.jsx:
```diff
@@ -5,7 +5,7 @@ import dispatcher from '../dispatcher';
  * This function fetches all News Sources
```
fluxActions does not really describe what this file does, you might want to rename it and make it more descriptive

aanchirinah replied:
Ok I will do some refactoring

olawalejarvis commented on app/action/fluxActions.jsx:
```diff
@@ -5,7 +5,7 @@ import dispatcher from '../dispatcher';
  * This function fetches all News Sources
  * @return {[type]} [description]
```
You should specify what data type you are expecting this function to return in the Jdoc

olawalejarvis commented on app/action/fluxActions.jsx:
```diff
@@ -18,45 +18,62 @@ export function fetchAllNewsSources() {
-  } else {
-    axios.get(requestUrlLatest).then(
+export const fetchAllArticles = (newsSourcesID, sort = 'top') => {
+  const encodednewsSourcesID = encodeURIComponent(newsSourcesID);
```
This should be in camelCase like `encodedNewsSourcesid`

aanchirinah replied:
Will be fixed

olawalejarvis commented on app/tests/components/HeadlineSearch.test.jsx:
```diff
@@ -29,6 +29,10 @@ describe('Headline Search', () => {
+   });
   // it('should call onSearch with entered input text',()=>{
   //   const searchText = 'AlJazeera';
   //   const spy = expect.createSpy();
```
Remove commented codes

Avuidrauxs committed:
feat(flux tests): implement more flux tests
- fixed ui issues
- added more compnent tests
[delivers]

aanchirinah committed:
Merge pull request #16 from andela-aanchirinah/checkpoint/feedback/add-more-unit-and-behaviour-tests

Checkpoint 1.5 Review

Avuidrauxs committed:
Fixed minor errors

aanchirinah committed:
Merge pull request #18 from andela-aanchirinah/checkpoint/peer-review/feedback

Fixed minor errors

Avuidrauxs committed:
feat(feedback): implement feedbacks given on bad news app
- added more tests to improve test coverage
[delivers]

Avuidrauxs committed:
feat(feedback): implement feedbacks given on bad news app
- added more tests to improve test coverage
[delivers]

Avuidrauxs committed:
feat(feedback): implement feedbacks given on bad news app
- added more tests to improve test coverage
[delivers]

Avuidrauxs committed:
feat(tests):add more test coverage
[delivers]

Avuidrauxs committed:
feat(tests):add more test coverage
[delivers]

Avuidrauxs committed:
feat(tests):add more test coverage
[delivers]

Avuidrauxs committed:
fix(coverage): fixed covergae issues
[delivers]

Avuidrauxs committed:
feat(tests):add more test coverage
[delivers]

Avuidrauxs committed:
fix(Readme.md): fixed coveralls badge
- Fixed the coveralls branch badge url to show 'develop' branch
- Included and features paragraph
[fixed]

aanchirinah committed:
Merge pull request #19 from andela-aanchirinah/checkpoint/feedback2

Checkpoint 1.7 Review

Avuidrauxs committed:
fix(package.json): fix a typo
[fixed]

Avuidrauxs committed:
fix(package.json): fix a typo
[fixed]

Avuidrauxs committed:
fixed coverage

Avuidrauxs committed:
fixed coverage

Avuidrauxs committed:
feat(favourites): implement add and remove favourite for users per session
- included toasts to notify user if favourite is added and removed
- stores favourties locally per session
[delivered]

Avuidrauxs committed:
fix(all): implement checkpoint scorecard feedback
- Fixed adding and removing favourites bug
- increased texst coverage
- added more test for actions and stores
- removed commented codes in tests
- added more jsdoc for more files
- fixed eslint errors
- refacored some code for better variable naming
- fixed browser console errors emanating from 'NewsHome.jsx'
[fixed]

aanchirinah committed:
Merge pull request #20 from andela-aanchirinah/checkpoint/scorecard/feedback

Checkpoint 1.8 Review

Avuidrauxs committed:
fix(tests): fix eslint errors
- fix eslint error in all **.test.js files
[fixed]

aanchirinah committed:
Merge pull request #22 from andela-aanchirinah/checkpoint/scorecard-feedback/fix-tests-eslint

Checkpoint 1.9 Review

Avuidrauxs committed:
fix(tests): fix eslint errors
- fix eslint error in all **.test.js files
[fixed]

aanchirinah committed:
Merge pull request #23 from andela-aanchirinah/checkpoint/scorecard-feedback/fix-tests-eslint

Checkpoint 1.10 Review

aanchirinah committed:
Fixed jQuery vulnerability bug

aanchirinah committed:
Merge pull request #24 from aanchirinah/fix-vulneralbility-issue

Fixed jQuery vulnerability bug
